composeml/label_times.py

```diff
@@ -22,20 +30,44 @@ def __init__(self, data=None, name=None, target_entity=None, settings=None, tran
         self.transforms = transforms or []
         self.plot = LabelPlots(self)
 
+        if label_type is not None:
+            error = 'label type must be "continuous" or "discrete"'
+            assert label_type in ['continuous', 'discrete'], error
+
+        if label_type is None and name in self.columns:
+            label_type = self.infer_type()
+
+        self.label_type = label_type
+        self.settings['label_type'] = self.label_type
+
     @property
     def _constructor(self):
         return LabelTimes
 
     @property
-    def is_categorical(self):
-        """Whether labels are categorical."""
+    def is_discrete(self):
+        """Whether labels are discrete."""
         dtype = self[self.name].dtype
-        return pd.api.types.is_categorical_dtype(dtype)
+
+        is_discrete = pd.api.types.is_bool_dtype(dtype) \
+            or pd.api.types.is_categorical_dtype(dtype) \
+            or pd.api.types.is_object_dtype(dtype)
+
+        if is_discrete:
+            return True
+
+        labels = self[self.name].iloc[:100]
```

Comment:
for now, let's only look at the dtype to infer type. we can always add this functionality in later

Reply:
updated function